src/power_grid_model/core/power_grid_dataset.py

```diff
@@ -375,7 +389,7 @@ def __init__(self, dataset_ptr: WritableDatasetPtr):
         self._schema = power_grid_meta_data[self._dataset_type]
 
         self._component_buffer_properties = self._get_buffer_properties(info)
-        self._data: Dict[str, Union[np.ndarray, Dict[str, np.ndarray]]] = {}
+        self._data: Dict[ComponentType, Union[np.ndarray, Dict[str, np.ndarray]]] = {}
```

Comment:
```suggestion
        self._data: Dataset = {}
```
Or maybe check if it is some other dataset type in the data_types.py